libc/stdio_scanf.c

```diff
@@ -0,0 +1,4210 @@
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vfscanfWrapper(filep, "%[1-9] %[^1-9] %[1-9]", res1, res2, res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vsscanfWrapper(buff2, "%[1-9] %[^1-9] %[1-9]", res1, res2, res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(buff2, "%[1-9] %[^1-9] %[1-9]", res1, res2, res3));
+	TEST_ASSERT_EQUAL_STRING("123", res1);
+	TEST_ASSERT_EQUAL_STRING("loremIPSUM IPSUMdolor", res2);
+	TEST_ASSERT_EQUAL_STRING("123", res3);
+}
+
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(stdio_scanf_rest)
+{
+	filep = fopen(TESTFILE_PATH, "w+");
+}
+
+
+TEST_TEAR_DOWN(stdio_scanf_rest)
+{
+	fclose(filep);
+}
+
+
+TEST(stdio_scanf_rest, modifiers_mix)
+{
+	int int1, int2, int3, int4, int5;
+	unsigned int res2;
+	float flt1, flt2, flt3;
+	long long int llint1;
+	char char1;
+	ptrdiff_t ptr1;
+	char str1[MAX_TESTSTR_WORDLEN] = { 0 };
+	char str2[MAX_TESTSTR_WORDLEN] = { 0 };
+	char str3[MAX_TESTSTR_WORDLEN] = { 0 };
+	char str4[MAX_TESTSTR_WORDLEN] = { 0 };
+	const char *str = "~~1`2l 0.1!_2@lorem#0x233$ 1.2e-5 % nowy 1.200020e-5 nal^ 132 *{}:|?><[]',./5/123456/+123456-a(loremipsum\0)";
+	char *format = "~~%d`%ul %f!_%x@%[lorem]#%p$%a %% %s %e nal^ %i *{}:|?><[]',./%o/%lld/+%hhx-%c(%[^ipsum]%s)";
+
+	fprintf(filep, "%s", TEST_STR);
+	rewind(filep);
+
+	sscanf(str, format, &int1, &res2, &flt1, &int2, str1, &ptr1, &flt2, str2, &flt3, &int3, &int4, &llint1, &int5, &char1, str3, str4);
+
+	TEST_ASSERT_EQUAL_INT(1, int1);
+	TEST_ASSERT_EQUAL_UINT(2, res2);
+	TEST_ASSERT_EQUAL_FLOAT(0.1, flt1);
+	TEST_ASSERT_EQUAL_HEX(2, int2);
+	TEST_ASSERT_EQUAL_STRING("lorem", str1);
+	TEST_ASSERT_EQUAL_INT64(563, ptr1);
+	TEST_ASSERT_EQUAL_FLOAT(1.200000e-05, flt2);
+	TEST_ASSERT_EQUAL_STRING("nowy", str2);
+	TEST_ASSERT_EQUAL_FLOAT(1.200020e-05, flt3);
+	TEST_ASSERT_EQUAL_UINT(132, int3);
+	TEST_ASSERT_EQUAL_INT(5, int4);
+	TEST_ASSERT_EQUAL_INT64(123456, llint1);
+	TEST_ASSERT_EQUAL_HEX8(0x56, int5);
+	TEST_ASSERT_EQUAL_CHAR('a', char1);
+	TEST_ASSERT_EQUAL_STRING("lore", str3);
+	TEST_ASSERT_EQUAL_STRING("mipsum", str4);
+}
+
+
+TEST(stdio_scanf_rest, m_s)
+{
+/* Disabled because of issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/667 */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+
+	const char *lorem = "LoremIpsumDolorSitAmet,Vestibulum";
+	char *res;
+	char chrArray[BUFF_LEN];
+	int i;
+
+	for (i = 0; i < BUFF_LEN - 1; i++) {
+		chrArray[i] = 'a';
+		if (i == BUFF_LEN - 5) {
+			chrArray[i] = 'b';
```

Comment:
Why do we want to have this 'b' inside the tested string? In my opinion not needed.

```suggestion
		if (i == BUFF_LEN - 5) {
			chrArray[i] = 'b';
```